### ISSUE TITLE
Does this lib work with storyboards?

### ISSUE THREAD

**Issue by pawelkata** [henrytkirk/HTKDragAndDropCollectionViewLayout#5]
I can't dequeue a cell from a storyboard setup and registering cell during configuration does not seem to work either.

**Comment by henrytkirk** [comment]
Can you show me some sample code?

**Comment by pawelkata** [comment]
I'm not doing anything special... just the regular:

```
- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
  return 1;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
  return _counters.count;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
  CounterCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"counterCell" forIndexPath:indexPath];
  cell.lblName.text = _counters[indexPath.row];
  cell.lblAmount.text = @"20";
  return cell;
}
```

The above setup works well, but when I inherit from classes in this lib (everything as in the demo app you provide), the app throws an assert error on the dequeue.

**Comment by henrytkirk** [comment]
If you are using Storyboard - are you registering the Nib to counterCell? You need to do something like this (except you want registerNib:…)

    [self.collectionView registerClass:[MyCellClass class] forCellWithReuseIdentifier:@“counterCell"];

**Comment by pawelkata** [comment]
As far as I know it isn't needed with storyboards when you set the reuse identifier. Nevertheless I did try registering my collectionViewCell class - it stopped the assert, but nothing showed up (not even a dummy cell), despite the proper setup, which works on regular classes. I debugged the code and cell is being set up, but nothing shows up in the end.

**Comment by henrytkirk** [comment]
I didn’t test this with Storyboard (not a huge fan of it). I’ll do it shortly and let you know. It’s very possible there’s a bug I need to work out.
- Henry

**Comment by pawelkata** [comment]
No worries. Please do let me know when you update the code. I'll be glad to test it on my end and report results.
- Pawel

**Comment by DZamataev** [comment]
Hello!
Tried the storyboard today.
Creating dynamic prototype in storyboard won't do the work. I suppose it's a bug.
But registering the nib works.
```
- (void)viewDidLoad {
    [super viewDidLoad];    
    [self.collectionView registerNib:[UINib nibWithNibName:@"SampleCell" bundle:nil]
          forCellWithReuseIdentifier: @"SampleCell"];
```
and another bug is that ```HTKDraggableCollectionViewCell``` class does not call ```[self setupDraggableCell];``` in ```-(void)awakeFromNib``` method. It does not implement it.
So the workaround is to implement it in a subclass of cell like this:
```
- (void)awakeFromNib {
    [super awakeFromNib];
    // next line fixes the absense of awakeFromNib implementation in parent class by performing the selector which must be performed after init to setup the cell
    if ([super respondsToSelector:NSSelectorFromString(@"setupDraggableCell")]) {
        [super performSelector:NSSelectorFromString(@"setupDraggableCell") withObject:nil];
    }
}
```

Hope that helps!
Thanks for the contribution!

**Comment by pawelkata** [comment]
Thanks for your insights, @DZamataev.

**Comment by henrytkirk** [comment]
Sorry for delay with this - I will push a basic fix adding awakeFromNib. Should solve the storyboard/nib issue as noted above.

**Comment by henrytkirk** [comment]
Fixed in commit: 9619a47e68611df5dd803a0c4aee66efd40a42a3
